apps/prairielearn/src/lib/socket-server.js

```diff
@@ -1,67 +1,58 @@
 // @ts-check
 const { Server } = require('socket.io');
-const redis = require('redis');
+const { Redis } = require('ioredis');
 const { createAdapter } = require('@prairielearn/socket.io-redis-adapter');
 const { logger } = require('@prairielearn/logger');
 const path = require('path');
 const debug = require('debug')('prairielearn:' + path.basename(__filename, '.js'));
 
 const { config } = require('./config');
 
+/**
+ * @param {Redis} client
+ * @param {string} type
+ */
+function attachEventListeners(client, type) {
+  client.on('error', (err) => {
+    logger.error(`redis ${type} client: error`, err);
+  });
+  client.on('connect', () => {
+    logger.verbose(`redis ${type} client: connect`);
+  });
+  client.on('ready', () => {
+    logger.verbose(`redis ${type} client: ready`);
+  });
+  client.on('reconnecting', () => {
```

Comment:
Apparently this [takes an argument](https://github.com/redis/ioredis#connection-events):
> The argument of the event is the time (in ms) before reconnecting.